addons/mail/static/src/model/model_field.js

```diff
@@ -352,103 +352,78 @@ class ModelField {
      * @param {mail.model} record
      * @param {any} newVal
      * @param {Object} [options]
-     * @returns {boolean} whether the value changed for the current field
-     */
-    parseAndExecuteCommands(record, newVal, options) {
-        if (newVal instanceof FieldCommand) {
-            // single command given
-            return newVal.execute(this, record, options);
-        }
-        if (newVal instanceof Array && newVal[0] instanceof FieldCommand) {
-            // multi command given
-            let hasChanged = false;
-            for (const command of newVal) {
-                if (command.execute(this, record, options)) {
-                    hasChanged = true;
-                }
-            }
-            return hasChanged;
-        }
-        // not a command
-        return this.set(record, newVal, options);
-    }
-
-    /**
-     * Get the raw value associated to this field. For relations, this means
-     * the local id or list of local ids of records in this relational field.
-     *
-     * @param {mail.model} record
-     * @returns {any}
-     */
-    read(record) {
-        return record.__values[this.fieldName];
-    }
-
-    /**
-     * Set a value on this field. The format of the value comes from business
-     * code.
-     *
-     * @param {mail.model} record
-     * @param {any} newVal
-     * @param {Object} [options]
      * @param {boolean} [options.hasToUpdateInverse] whether updating the
      *  current field should also update its inverse field. Only applies to
      *  relational fields. Typically set to false only during the process of
      *  updating the inverse field itself, to avoid unnecessary recursion.
      * @returns {boolean} whether the value changed for the current field
      */
-    set(record, newVal, options) {
+    parseAndExecuteCommands(record, newVal, options) {
         const currentValue = this.read(record);
-        if (this.fieldType === 'attribute') {
+        if (this.fieldType === 'attribute' && !(newVal instanceof FieldCommand)){
             if (currentValue === newVal) {
                 return false;
             }
             record.__values[this.fieldName] = newVal;
             return true;
         }
-        if (this.fieldType === 'relation') {
-            let hasChanged = false;
-            for (const val of newVal) {
-                switch (val[0]) {
+
+        let hasChanged = false;
+        if (!(newVal instanceof Array)) {
+            newVal = [newVal];
+        }
+        for (const val of newVal) {
+            const name = val instanceof FieldCommand ? val.name : val[0];
+            const data = val instanceof FieldCommand ? val.data: val[1];
```

Comment:
I would rename `name` to `commandName`, and `data` to something more explicit but I don't have a better idea right now. I guess maybe it should be called `newVal`, and the other `newVal` renamed to something like `commandList`?